{{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py

```diff
@@ -10,56 +10,99 @@
 {%- set widget_name = cookiecutter.plugin_name.replace(' ', '') + 'Widget' -%}
 {%- set config_page = cookiecutter.plugin_name.replace(' ', '') + 'ConfigPage' %}
 
+{%- if cookiecutter.graphical_plugin == 'n' %}
 from qtpy.QtWidgets import QVBoxLayout
-
-{% if cookiecutter.spyder3_compatibility == 'y' -%}
+{%- endif %}
+{% if cookiecutter.spyder3_compatibility == 'y' %}
 try:
     from spyder.api.plugins import SpyderPluginWidget
 except ImportError:
     from spyder.plugins import SpyderPluginWidget # Spyder3
     {%- if cookiecutter.create_config_page == 'y' %}
 try:
-    from spyder.api.preferences import
+    from spyder.api.preferences import PluginConfigPage
 except ImportError:
     from spyder.plugins.configdialog import PluginConfigPage # Spyder3
     {%- endif %}
-{% else -%}
+{% else %}
+    {%- if cookiecutter.graphical_plugin == 'y' %}
 from spyder.api.plugins import SpyderPluginWidget
+    {%- else %}
+from spyder.api.plugins import SpyderPlugin
+    {%- endif %}
     {%- if cookiecutter.create_config_page == 'y' %}
 from spyder.api.preferences import PluginConfigPage
-    {%- endif %}
-{%- endif -%}
-
+    {% endif %}
+{%- endif %}
 from .widgets.{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui import {{ widget_name }}
-{% if cookiecutter.create_config_page == 'y' %}
 
+{% if cookiecutter.create_config_page == 'y' %}
 class {{ config_page }}(PluginConfigPage):
     """{{ cookiecutter.plugin_name }} plugin preferences."""
     pass
 {% endif %}
+{%- if cookiecutter.graphical_plugin == 'n' %}
+class {{ cookiecutter.plugin_name.replace(' ', '') }}Plugin(SpyderPlugin):
+    """{{ cookiecutter.plugin_name }} plugin."""
+    CONF_SECTION = '{{ cookiecutter.plugin_name }}'
+    {% if cookiecutter.create_config_page == 'y' %}
+    CONFIGWIDGET_CLASS = {{ config_page }}
+    {% else %}
+    CONFIGWIDGET_CLASS = None
+    {% endif %}
+    def __init__(self, parent=None):
+        SpyderPlugin.__init__(self, parent)
+
+        # Create widget and add to dockwindow
+        self.widget = {{ widget_name }}(self.main)
+
+        # Initialize plugin
+        self.initialize_plugin()
+
+    # --- SpyderPlugin API ----------------------------------------------
+    def get_plugin_title(self):
+        """Return widget title."""
+        return "{{ cookiecutter.plugin_name }}"
 
+    def refresh_plugin(self):
+        """Refresh {{ widget_name }} widget."""
+        pass
+
+    def get_plugin_actions(self):
+        """Return a list of actions related to plugin."""
+        return []
+
+    def register_plugin(self):
+        """Register plugin in Spyder's main window."""
+        pass
+
+    def on_first_registration(self):
+        """Action to be performed on first plugin registration."""
+        pass
+
+    def apply_plugin_settings(self, options):
+        """Apply configuration file's plugin settings."""
```

Comment:
None of these methods (i.e. get_plugin_title, refresh_plugin, etc) are required for non-graphical plugins. Please see our API for SpyderPlugin here:

https://github.com/spyder-ide/spyder/blob/master/spyder/api/plugins.py#L138